Sources/Tracing/Tracer.swift

```diff
@@ -315,12 +315,42 @@ public func withSpan<T>(
 ///   - instant: the time instant at which the span started
 ///   - context: The `ServiceContext` providing information on where to start the new ``Span``.
 ///   - kind: The ``SpanKind`` of the new ``Span``.
+///   - isolation: Defaulted parameter for inheriting isolation of calling actor
 ///   - function: The function name in which the span was started
 ///   - fileID: The `fileID` where the span was started.
 ///   - line: The file line where the span was started.
 ///   - operation: The operation that this span should be measuring
 /// - Returns: the value returned by `operation`
 /// - Throws: the error the `operation` has thrown (if any)
+#if swift(>=6.0)
```

Comment:
Is `#if swift(>=6.0)` correct here? Should this use `#if compiler(>=6.0.0)` instead, if a Swift 6 compiler in 5.x mode understands `#isolation`? I just tried to adopt this in pre-6 code with a 6.0 compiler and it was using the previous overload.